### PR TITLE
Allow model attributes to be hidden/visible

### LINKED ISSUE #119: Improve model to_array method
Instead of casting the model to an array straight (and exposing potentially private information such as post password) provide a map and/or better support for mapping of model attributes.

https://laravel.com/docs/8.x/eloquent-serialization

**Acceptance Criteria**
- Include model appends (`append`, `setAppend`, `makeVisible`, `makeHidden`)

### PR DESCRIPTION
**Pull request by srtfisher**
Improves support for model serialization. Closes #119.

### CODE REVIEW

File: src/mantle/framework/database/model/class-model.php [hunk 1]
```diff
@@ -28,12 +28,12 @@
  * @mixin \Mantle\Framework\Database\Query\Builder
```
Comment (efuller) [inline src/mantle/framework/database/model/class-model.php:28]: Looks like you could remove this `@todo` now https://github.com/alleyinteractive/mantle-framework/pull/139/files#diff-1273adc6b762163f4f03cd56c4750f3d2918d3c031918553300a61ef4b906ddfR26

File: src/mantle/framework/database/model/concerns/trait-has-attributes.php [hunk 2]
```diff
@@ -132,6 +143,57 @@ public function get_attributes(): array {
...
+	 *
+	 * @param string[] $values Values to check.
+	 * @return array
+	 */
+	protected function get_arrayable_items( array $values ): array {
+		$visible = $this->get_visible();
+		$hidden  = $this->get_hidden();
+
+		if ( ! empty( $visible ) ) {
+			$values = array_intersect_key( $values, array_flip( $visible ) );
```
Comment (efuller) [inline src/mantle/framework/database/model/concerns/trait-has-attributes.php:187]: `array_flip` neat!

**Review by efuller — APPROVED** [review]
🛡️

File: src/mantle/framework/database/model/class-model.php (hunk 1 above)
Comment (srtfisher) [reply src/mantle/framework/database/model/class-model.php:26]: good catch!

**Pull request merged**